R/reshape_grouplevel.R

```diff
@@ -1,80 +1,76 @@
+#'   `"Participant"`, if the model was `y ~ x + (1|Participant)`.
+#'
 #' @export
-reshape_grouplevel <- function(x, indices = "all", group = "all", ...) {
+reshape_grouplevel <- function(x, ...) {
   UseMethod("reshape_grouplevel")
 }
 
+#' @export
+reshape_grouplevel.default <- function(x, ...) {
+  insight::format_error(paste0(
+    "`reshape_grouplevel()` not implemented yet for objects of class `",
+    class(x)[1],
+    "`."
+  ))
+}
 
+#' @rdname estimate_grouplevel
 #' @export
-reshape_grouplevel.estimate_grouplevel <- function(x, indices = "all", group = "all", ...) {
+reshape_grouplevel.estimate_grouplevel <- function(x, indices = "all", group = NULL, ...) {
   # Find indices
   if (any(indices == "all")) {
     indices <- names(x)[!names(x) %in% c("Group", "Level", "Parameter", "CI")]
   }
+
+  # Accommodate Bayesian
   if ("Coefficient" %in% indices) {
-    indices <- c(indices, "Median", "Mean", "MAP") # Accommodate Bayesian
+    indices <- c(indices, "Median", "Mean", "MAP")
   }
   if ("SE" %in% indices) {
-    indices <- c(indices, "SD", "MAD") # Accommodate Bayesian
+    indices <- c(indices, "SD", "MAD")
   }
-  indices <- names(x)[names(x) %in% unique(indices)]
+  indices <- intersect(colnames(x), unique(indices))
 
   # Random parameters
-  if (all(group == "all")) group <- unique(x$Group)
-
-  # Loop through all groups
-  for (g in group) {
-    # Subset coefficients
-    data_group <- x[x$Group == g, ]
-
-    # If no data, skip
-    if (nrow(data_group) == 0) next
-
-    # Clean subset of random factors
-    data_group[[g]] <- data_group$Level
-    newvars <- paste0(g, "_", indices)
-    names(data_group)[names(data_group) %in% indices] <- newvars
-
-    # Reshape
-    data_group$Parameter <- ifelse(data_group$Parameter == "(Intercept)",
-      "Intercept",
-      data_group$Parameter
-    )
-
-    data_wide <- datawizard::data_to_wide(
-      data_group[c(g, newvars, "Parameter")],
-      id_cols = g,
-      values_from = newvars,
-      names_from = "Parameter",
-      names_sep = "_"
-    )
+  if (is.null(group)) {
+    group <- unique(x$Group)
+  }
 
-    # If nested, separate groups
-    if (grepl(":", g, fixed = TRUE)) {
-      groups <- as.data.frame(t(sapply(strsplit(data_wide[[g]], ":", fixed = TRUE), function(x) as.data.frame(t(x)))))
-      names(groups) <- unlist(strsplit(g, ":", fixed = TRUE))
-      data_wide <- cbind(groups, data_wide)
-      data_wide[g] <- NULL
-      g <- names(groups)
-    }
+  if (length(group) > 1) {
+    insight::format_alert(paste0(
+      "Multiple groups are present (", toString(group),
+      "). Selecting the first (", group[1], ")."
+    ))
+    group <- group[1]
+  }
+  x <- x[x$Group %in% group, ]
 
-    # Merge while preserving order of original random
-    data <- attributes(x)$data # Get original dataframe of random
-    data[["__sort_id"]] <- seq_len(nrow(data))
-    data <- merge(data, data_wide, by = g, sort = FALSE)
-    data <- data[order(data[["__sort_id"]]), ]
```

Comment:
I think we still need something like this `merge()` in the code, to match the wide table, which is shorter than the original data, to the original data.

Reply:
I don't think we need to match to the original data now, otherwise it kind of disregards the `group` that was selected 

In other words, the resulting dataframe should match the number of individual items in the requested `group` (and we can leave it to the user to merge it to the original dataframe if need be)